src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -19,6 +19,7 @@
 using Raven.Server.Documents.Includes;
 using Raven.Server.Documents.Indexes.Auto;
 using Raven.Server.Documents.Indexes.MapReduce.Auto;
+using Raven.Server.Documents.Indexes.MapReduce.Static;
```

Comment:
The idea was to define the indexing fields that we already know (based on indexing unctions) and include them in the index definition (they there should be there anyway IMO). Next, when we know what is the index output, we can easily pass that info to a Lucene doc converter - the same as we do for auto indexes. Just wanted to have the same conversion approach for all kind of indexes (auto/static/map/map-reduce) that is to relay on fields from the index definition.

But you are right - we cannot do it for dynamic fields. So if an index uses `CreateFields` method then `# output fields != # fields in def` and we need to index additional fields - current code does not support that, I will make sure it is properly handled.